exporter/azure_monitor/azure_monitor.go

```diff
@@ -20,18 +21,13 @@ type AzureTraceExporter struct {
 	@param options holds specific attributes for the new exporter
 	@return The exporter created and error if there is any
 */
-func NewAzureTraceExporter(IKey string) (*AzureTraceExporter, error) {
-	if IKey == "" {
+func NewAzureTraceExporter(Options common.Options) (*AzureTraceExporter, error) {
+	if Options.InstrumentationKey == "" {
 		return nil, errors.New("missing Instrumentation Key for Azure Exporter")
 	}
-	currentOptions := common.Options {
-		InstrumentationKey: IKey,
-		EndPoint:           "https://dc.services.visualstudio.com/v2/track",
-		TimeOut: 			10.0,
-	}
 	exporter := &AzureTraceExporter {
-		InstrumentationKey: currentOptions.InstrumentationKey,
-		Options:            currentOptions,
+		InstrumentationKey: Options.InstrumentationKey,
+		Options:            Options,

```

Comment:
Still seems kind of weird to have iKey in Options and you passing in both an iKey and an Options object. Is there a way in GoLang to pass in only the iKey as mandatory and use default values for the other options if they are not specified?

Reply:
Yeah, I updated it. This should be better.

Reply:
Nice this is good. I don't think you need InstrumentationKey anymore in AzureTraceExporter